File: main_app/android/app/src/main/java/com/lifestyle/dailyscript/data/model/UserPrefs.kt
package com.lifestyle.dailyscript.data.model

import kotlinx.serialization.Serializable

/**
 * 온보딩에서 고른 선호도 (PWA ds.pref / users.pref_* 컬럼과 동일 값).
 *  - genres: 작품 format 값들 ("novel","play","essay","opera","prose")
 *  - themes: 10범주 한글 주제명 — PWA 저장값과 문자열이 정확히 일치해야 기기 간 동기화가 된다.
 *  - any: "상관없음"(모든 주제 폭넓게) 선택 여부
 */
@Serializable
data class UserPrefs(
    val genres: List<String> = emptyList(),
    val themes: List<String> = emptyList(),
    val any: Boolean = false,
) {
    /** 실제로 추천을 좁히는 선호가 있나? (PWA hasActivePrefs) */
    fun hasActive(): Boolean = genres.isNotEmpty() || (!any && themes.isNotEmpty())
}

File: main_app/ios/Curtaincall/Curtaincall/Models/UserPrefs.swift
import Foundation

/// Onboarding preferences chosen on first launch. Mirrors Android `UserPrefs`
/// and the PWA's `ds.pref`.
///  - `genres`: works.format values ("novel","play","essay","opera","prose").
///  - `themes`: the 10 `CardTheme` category names — must match the classifier's
///    category strings exactly so a saved theme actually weights cards.
///  - `any`: "상관없음" — recommend broadly across all themes.
///
/// Stored locally (UserDefaults) for now. 이창훈 said `users.pref_*` columns are
/// planned ("DB칼럼으로 쓸겁니다") but may not be live yet, so the DB write is
/// deliberately omitted until those columns are confirmed deployed.
struct UserPrefs: Equatable, Sendable {
    var genres: [String]
    var themes: [String]
    var any: Bool

    init(genres: [String] = [], themes: [String] = [], any: Bool = false) {
        self.genres = genres
        self.themes = themes
        self.any = any
    }

    /// Whether these prefs actually narrow recommendations (Android `hasActive`).
    var hasActive: Bool { !genres.isEmpty || (!any && !themes.isEmpty) }
}
